docs/how-tos/how-to-use-searchkick-for-search-autocomplete.mdx

```diff
@@ -0,0 +1,285 @@
+    if params[:keywords].present?
+-       Spree::Taxon
+-         .where(Spree::Taxon.arel_table[:name].matches("%#{params[:keywords]}%"))
+-         .limit(5)
++       Spree::Taxon.search(params[:keywords], fields: [{name: :word_start}], limit: 5)
+    else
+      Spree::Taxon.none
+    end
+  end
+```
+
+With just these small changes, the autcomplete will finally use Searchkick
+under the hood.
+
+
+## Next steps
+
+We just stratched the surface of what's possible with Searchkick, and
+it's left to each store to implement the configuration that better suits
+its needs. Still, there are some big things worth considering and in
+this section we'll see the most relevant for a typical Solidus store.
+
+:::info
+
+[Searchkick README][searchkick-github] is really well done and contains
+more details about the information reported in this guide. Please, always
+refer to that documentation to get extended and updated information.
+
+:::
+
+
+### Reindex Strategies
+
+By default, everytime you change a record, if Searchkick is enabled
+on that model, a reindex will be triggered for that record only.
+While this is very useful, it can hurt stores perfomances.
+
+To mitigate this problem, it is suggested to use the Asynchronous
+Reindex Strategy, which will push the update to a background job.
+
+To enabale this strategy, you can edit the directive that enabled
+Searchkick on the models, like this:
+
+```diff title="app/overrides/my_store/spree/product/add_searchkick.rb"
+-   base.searchkick word_start: [:name]
++   base.searchkick word_start: [:name], callbacks: :async
+```
+
+### Reindex Associations
+
+Another important aspect of the the automatic reindexing is how
+associations are reindexed. In fact, data is not automatically
+synced when an association is updated. This is useful if the
+information indexed for a given record depends on the state of
+another resource of your application, associated with that record.
+
+For example, imagine if you only want to index products that has
+stock in your warehouses. You can easily add this behavior using
+the `should_index?` method as we saw before:
+
+
+```diff title="app/overrides/my_store/spree/product/add_searchkick.rb"
+    def should_index?
+-     kept?
++     kept? && total_on_hand > 0
+    end
+```
+
+:::info
+
+`total_on_hand` checks all the product's variants stock levels via a
+model called Spree::StockItem. This model contains all the information
+about the remaining stock for a given variant in a given warehose.
+
+:::
+
+Nice and clean, but what if someone purchase that product and its stock
+goes to 0?
+
+We need to also tell Solidus to reindex products associated with the
+corresponding StockItem, every time they get an update.
+
+```ruby title="app/overrides/my_store/spree/stock_item/reindex_product_on_change.rb"
+module AmazingStore
+  module Spree
+    module StockItem
+      module ReindexProductOnChange
+        def self.prepended(base)
+          base.after_commit :reindex_product
```

Comment:
I wouldn't say I like encouraging the use of `after_commit` callbacks, but not sure if there's any other way at this point.

Reply:
Yes, that would be great if we have an event emitted or a service to create/update products. Also, this is what's suggested in the Searchkick [Associations](https://github.com/ankane/searchkick#associations) part of the guide, and I think there's some value in keeping the guides similar. Open to other ideas, as always.